src/instance-config.ts

```diff
@@ -0,0 +1,496 @@
+   *
+   * @see {@link v1.InstanceAdminClient#updateInstanceConfig}
+   * @see [UpdateInstanceConfig API Documentation](https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.admin.instance.v1#google.spanner.admin.instance.v1.InstanceAdmin.UpdateInstanceConfig)
+   *
+   * @param {object} metadata The metadata you wish to set.
+   * @param {object} [gaxOptions] Request configuration options,
+   *     See {@link https://googleapis.dev/nodejs/google-gax/latest/interfaces/CallOptions.html|CallOptions}
+   *     for more details.
+   * @param {SetInstanceConfigMetadataCallback} [callback] Callback function.
+   * @returns {Promise<LongRunningOperationResponse>}
+   *
+   * @example
+   * ```
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   *
+   * const instanceConfig = spanner.instanceConfig('custom-my-instance-config');
+   *
+   * const metadata = {
+   *   displayName: 'My Instance Config'
+   * };
+   *
+   * instanceConfig.setMetadata(metadata, function(err, operation, apiResponse) {
+   *   if (err) {
+   *     // Error handling omitted.
+   *   }
+   *
+   *   operation
+   *     .on('error', function(err) {})
+   *     .on('complete', function() {
+   *       // Metadata updated successfully.
+   *     });
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * instanceConfig.setMetadata(metadata).then(function(data) {
+   *   const operation = data[0];
+   *   const apiResponse = data[1];
+   * });
+   * ```
+   */
+  setMetadata(
+    config: SetInstanceConfigMetadataRequest,
+  ): Promise<SetInstanceConfigMetadataResponse>;
+  setMetadata(
+    config: SetInstanceConfigMetadataRequest,
+    callback: SetInstanceConfigMetadataCallback
+  ): void;
+  setMetadata(
+    config: SetInstanceConfigMetadataRequest,
+    callback?: SetInstanceConfigMetadataCallback
+  ): void | Promise<SetInstanceConfigMetadataResponse> {
+    const reqOpts = {
+      instanceConfig: extend(
+        {
+          name: this.formattedName_,
+        },
+        config.instanceConfig
+      ),
+      updateMask: {
+        paths: Object.keys(config.instanceConfig).map(snakeCase),
+      },
+      validateOnly: config.validateOnly,
+    };
+
+    // validateOnly need not be passed in if it is null.
+    if(reqOpts.validateOnly == null) delete reqOpts.validateOnly;
+
+    return this.request(
+      {
+        client: 'InstanceAdminClient',
+        method: 'updateInstanceConfig',
+        reqOpts,
+        gaxOpts: config.gaxOpts,
+        headers: this.resourceHeader_,
+      },
+      callback!
+    );
+  }
+
+  /**
+   * @typedef {array} DeleteInstanceConfigResponse
+   * @property {object} 0 The full API response.
+   */
+  /**
+   * @callback DeleteInstanceConfigCallback
+   * @param {?Error} err Request error, if any.
+   * @param {object} apiResponse The full API response.
+   */
+  /**
+   * Delete the instance config.
```

Comment:
Maybe add something like: This function can only be called for custom instance configs.

Reply:
Done.